src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -2735,42 +2761,68 @@ export default class AircraftModel {
         delete this.conflicts[conflictingAircraft.callsign];
     }
 
-    // TODO: needs better name
     /**
      * @for AircraftModel
-     * @method _contactAircraftAfterControllabilityChange
+     * @method _updateAircraftControllability
+     * @param isAutoTower {boolean}
      * @private
      */
-    _contactAircraftAfterControllabilityChange() {
-        // Crossing into the center
+    _updateAircraftControllability(isAutoTower) {
+        if (this.projected) {
+            return;
+        }
+
+        const newControllability = (isAutoTower && this._isUnderTowerControl()) ?
+            false : this.isInsideAirspace(AirportController.airport_get());
+
+        if (this.isControllable === newControllability) {
+            return;
+        }
+
+        this.isControllable = newControllability;
+
         if (this.isControllable) {
             this.callUp();
 
             return;
         }
 
         this.setIsRemovable();
-        EventBus.trigger(AIRCRAFT_EVENT.AIRSPACE_EXIT, this);
+        if (this.flightPhase !== FLIGHT_PHASE.LANDING) {
+            EventBus.trigger(AIRCRAFT_EVENT.AIRSPACE_EXIT, this);
+        }
     }
 
     /**
      * @for AircraftModel
-     * @method _updateAircraftControllability
+     * @method _isUnderTowerControl
      * @private
      */
-    _updateAircraftControllability() {
-        if (this.projected) {
-            return;
+    _isUnderTowerControl() {
+        if (_includes(AUTOTOWER_FLIGHT_PHASES, this.flightPhase)) {
+            return true;
+        }
+        if (this.category !== FLIGHT_CATEGORY.DEPARTURE || this.flightPhase !== FLIGHT_PHASE.CLIMB) {
+            return false;
         }
 
-        const isInsideAirspace = this.isInsideAirspace(AirportController.airport_get());
+        // FAA JO 7110.65, Para. 3-9-3-b (civil acft)
+        const runway = this.fms.departureRunwayModel;
+        return this.positionModel.distanceToPosition(runway.positionModel) < nm(runway.length) + 0.5;
+    }
 
-        if (this.isControllable === isInsideAirspace) {
-            return;
+    /**
+     * Respond to user changing the tower control (autotower) settings
+     * by adjusting controllability as needed *without* making radio calls
+     *
+     * @for AircraftModel
+     * @method onTowerControllerChange
+     * @param isAutoTower {boolean}
+     */
+    onTowerControllerChange(isAutoTower) {
+        if (this._isUnderTowerControl()) {
```

Comment:
This is a nit pick, but we/I have tried really hard to enforce exclusive if statements. Meaning, if a method or function doesn't have what we need to do the work, `return`. A function/method should end like the name says.

So in this case, we would change the guts of this method to:
```js
if (!this._isUnderTowerControl()) {
  return;
}

this.isControllable = !isAutoTower;
```

This gives us clear visibility into both logical branches and also makes the method read like the name

Reply:
done